### DISCUSSION TITLE
Issue when searching for Tags/Logs in the UI

### DISCUSSION THREAD

**Discussion by cptkng23** [jaegertracing/jaeger#3046] [Q&A]
Hi everyone.

I have a question around searching for Tags/Logs in the Jaeger UI. I have the feeling searching with wildcards/regex does not work if the _Log_ or _Tag_ content is longer than a certain threshold. I wasn't able to find any documentation around this (also not in github issues), so I wanted to confirm here.

For example, we have Spans with a _Log_ entry named `log1`, value is _hello_, and maximum length of all _Logs_ of those is 80 chars. Then we have a _Log_ entry named `log2`, one of the values is _world_, and the maximum length of all entries is 2kb.
Now searching with `log1=.*ell.*` in the UI returns results, but searching with `log2=.*orl.*` not.

How can I make the _log2_ field also searchable?

EDIT: We are using ES on AWS as storage backend.

PS:
- I'm not sure if this is a limitation (if any) in the UI or in the backend, therefore opening the discussion in this repo.
- Also, I have asked in [#jaeger](https://cloud-native.slack.com/archives/CGG7NFUJ3/p1619535601307200)  some time ago, but didn't get the answer I was looking for.

**Comment by jpkrohling** [comment]
@rubenvp8510, would you know what's going on?

**Reply by rubenvp8510** [reply]
I don't think there will be a limitation on the front end or in the query service, may be is an issue with elasticsearch? I'll try to reproduce this behaviour .

**Reply by cptkng23** [reply]
Thanks so much for checking this.

**Reply by cptkng23** [reply]
Hi @rubenvp8510 .

Were you by chance able to reproduce this?

**Reply by cptkng23** [reply]
Hi @rubenvp8510 
We are still facing this issue. Is there any more data/hints I can provide you with that would help in pinpointing the issue?

**Reply by rubenvp8510** [reply]
Hi @cptkng23 sorry for the delay on this. I still not able to reproduce this. There should not be any limitation on the query service or the UI. IIRC there is no limitation on the ES storage.

Could you verify that you indeed have spans with those log values stored on your ElasticSearch instance? 

I used this code to generate the spans with logs as you described in your initial comment: https://gist.github.com/rubenvp8510/d6b12a1fcbcff56af5126b99fe7cd9db

**Reply by cptkng23** [reply]
Thanks so much for checking. I'll have a deeper look myself on Monday.

**Reply by cptkng23** [reply]
Hi @rubenvp8510 
I checked everything twice. The data is there. I now used real data and didn't make it up with hello and world like in my first message.
For the example/screenshot below I use a trace with ID **e6041b5**.

First a successful search for a log field:
![log-search-returns-result](https://user-images.githubusercontent.com/8835232/128707222-8f58f7dd-9ec8-4a06-bfaa-8a6184ac79a2.png)
Below is a screenshot prooving that the data is also actually there.
![tracen-content-with-demandResponseHeaders](https://user-images.githubusercontent.com/8835232/128707464-6ea6315b-8357-4fcb-ac86-048f6662363c.png)

Now the one that doesn't work:
![log-search-no-result](https://user-images.githubusercontent.com/8835232/128707634-b071b567-a6ce-4a0a-93fa-f05f2b8f2492.png)
But below you can see that the data is actually there.
![trace-content-with-bidRequest](https://user-images.githubusercontent.com/8835232/128707652-e4019fd1-35dd-4fb2-ab0e-f08d6e813b6f.png)


I also downloaded this trace from the UI as json, and data is available. The size of the overall trace is ~30kb, and the length of the field demandResponseHeaders is 84kB, and the bidRequest is 1.9kB
![trace-json-snippet](https://user-images.githubusercontent.com/8835232/128708524-83b16465-da8f-4df0-a1c2-6c56eb453eb8.png)

If it's not about the size, maybe I'm missing something else here? Maybe the search term I use is wrong and just luckily returns results for the first query? Any help/hint is appreciated.

**Comment by cptkng23** [comment]
Could it be related to the mappings that have an ignore_above of 256? https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/es/mappings/jaeger-span.json#L84
